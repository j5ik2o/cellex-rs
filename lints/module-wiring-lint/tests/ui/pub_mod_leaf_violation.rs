#![feature(register_tool)]
#![register_tool(module_wiring)]
#![warn(module_wiring::no_parent_reexport)]

pub mod leaf {
  pub struct Leaf;
}

fn main() {}
