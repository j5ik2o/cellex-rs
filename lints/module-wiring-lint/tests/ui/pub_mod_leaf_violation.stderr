warning: 末端モジュールの宣言では `mod` のみを使用してください
  --> $DIR/pub_mod_leaf_violation.rs:5:1
   |
LL | / pub mod leaf {
LL | |   pub struct Leaf;
LL | | }
   | |_^
   |
   = note: ルール: 末端モジュールを公開したい場合は親で `mod` のみ宣言し、`pub use` で公開します
   = note: 詳細: 宣言 `pub mod leaf {
             pub struct Leaf;
           }` に公開可視性が設定されています
   = help: 宣言から可視性修飾子を削除し、公開が必要なシンボルは末端モジュール内で `pub` を付けてください
   = note: `#[warn(no_parent_reexport)]` on by default

warning: 1 warning emitted

